app-backend/tile/src/main/scala/TileIO.scala

```diff
@@ -0,0 +1,262 @@
+    * @param sceneId scene to request metadata for
+    * @param zoom level of metadata desired
+    * @return zoom level and metadata
+    */
+  def getSceneTileMetadata(sceneId: UUID, zoom: Int): OptionT[Future, (Int, TileLayerMetadata[SpatialKey])] = {
+
+    logger.debug(s"Requesting tile layer metadata (layer: $sceneId, zoom: $zoom")
+    rfCache.cachingOptionT(s"tile-metadata-$sceneId-$zoom")(getMaxZoomForScene(sceneId).mapFilter {
+      case (pyramidMaxZoom) =>
+        val layerName = sceneId.toString
+        for (maxZoom <- pyramidMaxZoom.get(layerName)) yield {
+          val z = if (zoom > maxZoom) maxZoom else zoom
+          blocking {
+            z -> store.readMetadata[TileLayerMetadata[SpatialKey]](LayerId(layerName, z))
+          }
+        }
+    })
+  }
+
+  /** Get tile for given extent and zoom level
+    *
+    * @param sceneId ID of scene
+    * @param zoom level to request
+    * @param extent extent of tile requested
+    * @return
+    */
+  def getSceneTileForExtent(sceneId: UUID, zoom: Int, extent: Extent): OptionT[Future, MultibandTile] = {
+    val cacheKey = s"extent-tile-$sceneId-$zoom-${extent.xmin}-${extent.ymin}-${extent.xmax}-${extent.ymax}"
+    OptionT(
+      timedFuture("layer-for-tile-extent-cache")(
+        rfCache.caching(cacheKey, doCache = cacheConfig.layerTile.enabled)( timedFuture("layer-for-tile-extent-s3")(
+          Future {
+            Try {
+              S3CollectionLayerReader(store)
+                .query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](LayerId(sceneId.toString, zoom))
+                .where(Intersects(extent))
+                .result
+                .stitch
+                .crop(extent)
+                .tile
+                .resample(256, 256)
+            } match {
+              case Success(tile) => Option(tile)
+              case Failure(e) =>
+                logger.error(s"Query layer $sceneId at zoom $zoom for $extent: ${e.getMessage}")
+                None
+            }
+          }
+        ))
+      )
+    )
+  }
+
+  /** Retrieve mosaic definitions -- usually the first step in constructing a mosaic
+    * for a project
+    *
+    * @param projectId project to request mosaic definition for
+    * @param polygonOption optional polygon to restrict mosaid definitions for
+    * @return
+    */
+  def getMosaicDefinition(projectId: UUID, polygonOption: Option[Projected[Polygon]]): OptionT[Future, Seq[MosaicDefinition]] = {
+    logger.debug(s"Reading mosaic definition (project: $projectId")
+    OptionT(timedFuture("get-mosaic-definition")(ScenesToProjects.getMosaicDefinition(projectId, polygonOption)))
+  }
+
+  /** Retrieve mosaic definitions -- usually the first step in constructing a mosaic
+    * for a project
+    *
+    * @param projectId project to request mosaic definition for
+    * @return
+    */
+  def getMosaicDefinition(projectId: UUID): OptionT[Future, Seq[MosaicDefinition]] = {
+    getMosaicDefinition(projectId, None)
```

Comment:
Kind of feels like `getMosaicDefinition`, `getMaxZoomForScene` and other functions which fetch scene/project metadata from the database should just hang on `PostgresAttributeStore` which would then become `RFAttributeStore`.